airflow/runtime_templates.go

```diff
@@ -0,0 +1,273 @@
+	}
+
+	return templateNames, nil
+}
+
+func SelectTemplate(templateList []string) (string, error) {
+	TemplatesTab := printutil.Table{
+		Padding:        []int{5, 30},
+		DynamicPadding: true,
+		Header:         []string{"#", "TEMPLATE"},
+	}
+	if len(templateList) == 0 {
+		return "", fmt.Errorf("no available templates found")
+	}
+
+	templateMap := make(map[string]string)
+
+	// Add rows for each template and index them
+	for i, template := range templateList {
+		index := i + 1
+		TemplatesTab.AddRow([]string{strconv.Itoa(index), template}, false)
+		templateMap[strconv.Itoa(index)] = template
+	}
+
+	TemplatesTab.Print(os.Stdout)
+
+	// Prompt user for selection
+	choice := input.Text("\n> ")
+	selected, ok := templateMap[choice]
+	if !ok {
+		return "", fmt.Errorf("invalid template selection")
+	}
+
+	return selected, nil
+}
+
+func InitFromTemplate(templateDir, destDir string) error {
+	HTTPClient := &httputil.HTTPClient{}
+	tarballURL := fmt.Sprintf("%s/tarball/main", astroTemplateRepoURL)
+
+	doOpts := &httputil.DoOptions{
+		Path:   tarballURL,
+		Method: http.MethodGet,
+	}
+
+	resp, err := HTTPClient.Do(doOpts)
+	if err != nil {
+		return fmt.Errorf("failed to download tarball: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("failed to download tarball, status code: %d", resp.StatusCode)
+	}
+
+	tempDir, err := os.MkdirTemp("", "extracted-")
+	if err != nil {
+		return fmt.Errorf("failed to create temp directory: %w", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Extract the tarball to the temporary directory
+	err = extractTarGz(resp.Body, tempDir, templateDir)
+	if err != nil {
+		return fmt.Errorf("failed to extract tarball: %w", err)
+	}
+
+	// Copy the extracted template directory to the destination directory
+	srcDir := filepath.Join(tempDir, templateDir)
+
+	if _, err := os.Stat(srcDir); os.IsNotExist(err) {
+		return fmt.Errorf("template directory %s not found", templateDir)
+	}
+	if err := copyDir(srcDir, destDir); err != nil {
+		return fmt.Errorf("failed to copy template directory: %w", err)
+	}
+
+	return nil
+}
+
+func extractTarGz(r io.Reader, dest, templateDir string) error {
+	gr, err := gzip.NewReader(r)
+	if err != nil {
+		return fmt.Errorf("failed to create gzip reader: %w", err)
+	}
+	defer gr.Close()
+
+	tarReader := tar.NewReader(gr)
+	var baseDir string
+
+	limitTarReader := io.LimitReader(tarReader, maxExtractSize)
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("failed to read tarball: %w", err)
+		}
+
+		// Skip over irrelevant files like pax_global_header
+		if header.Typeflag == tar.TypeXGlobalHeader {
+			continue
+		}
+
+		// Extract the base directory from the first valid header
+		if baseDir == "" {
+			parts := strings.Split(header.Name, "/")
+			if len(parts) > 1 {
+				baseDir = parts[0]
+			}
+		}
+
+		// Skip files that are not part of the desired template directory
```

Comment:
If we're already only extracting the files of the desired template, why do we need to extract it to a temp dir and then copy it all over to the final dir, instead of just extracting to the final dir?